### ISSUE TITLE
grunt serve:debug command is not working

### ISSUE THREAD

**Issue by charliemc** [angular-fullstack/generator-angular-fullstack#2332]
- [ ] I understand that GitHub issues are not for tech support, but for questions specific to this generator, bug reports, and feature requests.

Item | Version
----- | -----
generator-angular-fullstack | 3.3.0
Node | 4.4.0
npm | 3.8.1
Operating System | macOS Sierra 10.12.1

Item | Answer
----- | -----
DB | MongoDB

I'm having issues when trying to debug my app using the `grunt serve:debug` command. Initially, everything loads and node-inspector starts correctly stopping by default at the initial break-point.

Seconds after resuming the script execution, I get the following error on screen:
```
MongoDB connection error: MongoError: failed to connect to server [localhost:27017] on first connect
[nodemon] app crashed - waiting for file changes before starting...
```

![captura de pantalla 2016-10-30 a las 20 12 07](https://cloud.githubusercontent.com/assets/4527004/19839169/74d4a93a-9edd-11e6-8988-59b2bf821319.png)

It seems I am not being able to debug using `grunt serve:debug`, does anyone guess or know what could be causing this?

As additional information I have to say that everything else, other grunt tasks, such as `grunt serve`, `grunt test:server` have always worked fine. MongoDB is set up correctly and also works fine too.

**Comment by RonAlmog** [comment]
From what i can clearly see in your console, mongodb is not available. please check that.
does it run fine when you are not in debug mode? check your connection string.
OR, install Robomongo, and try to connect to your database and make sure it's up and running.
Another thing, if you haven't done that already, set your mongodb to run as a service, so you don't have to run in manually every time. that will save you from most of these issues.

**Comment by charliemc** [comment]
Hi @RonAlmog !

MongoDB is working fine, and I already use Robomongo and everything is working correctly. That's why I'm very confused at this point. Thank you for your response anyway!

**Comment by Awk34** [comment]
@charliemc Ron is right. The application can't connect to your local mongo daemon. You need to be running the mongo daemon locally on the default port.

**Comment by charliemc** [comment]
@Awk34 my mongo daemon is always running, and also running on the default port. In fact, I've been developing and working on this particular project that was generated from this generator for months now and never had any problems with mongo before. It's just when running `grunt serve:debug`.

There you have a screenshot, so you can see:

<img width="1541" alt="captura de pantalla 2016-11-03 a las 0 42 48" src="https://cloud.githubusercontent.com/assets/4527004/19951480/b57d2e52-a15e-11e6-9e41-f9f94713d421.png">

On the left, mongo daemon running on default port, on the right `grunt serve:debug`'s crash.

**Comment by thedewpoint** [comment]
@charliemc Did you have any luck fixing this? Just ran into this issue today.

**Comment by charliemc** [comment]
Hi @thedewpoint, I sadly didn't... I've tried to run this command from different machines and haven't been able to make it work.

**Comment by Awk34** [comment]
If you run `grunt serve`, everything works though? I find this all rather odd. Have you tried with a freshly-scaffolded application as well? (no modifications). You might also try using the latest version of the generator.

**Comment by thedewpoint** [comment]
I have a relatively new version, maybe only a few months old. Running gulp
serve regularly works perfect. Serve:debug starts up and then when it goes
to seed my db it can't connect. I've tried connecting to a remote mongodb
as well, and I've also tried different versions of mongo.

**Comment by Awk34** [comment]
@thedewpoint are you also on a Mac? Which MongoDB version are you running?

**Comment by thedewpoint** [comment]
I'm not at home now but it's windows 8 PC , not positive on the mongo
version but it's 2.x on my windows local, and my remote mongodb was running
on debian Linux and was 3.x neither worked

**Comment by charliemc** [comment]
@Awk34 yes, if I run `grunt serve` everything works perfectly. I'm about to start a new project based on this generator, I will try this command again on this new application and will let you know.

**Comment by thedewpoint** [comment]
Weird thing. I downloaded my project on my mac laptop and it works there.
Maybe this is a windows specific issue?

On Fri, Nov 11, 2016 at 10:37 AM, Charlie Martínez <notifications@github.com

**Comment by albert-92** [comment]
I got the same error on Windows 10 and generator-angular-fullstack 4.1.0

`MongoDB connection error: MongoError: failed to connect to server [localhost:27017] on first connect`

**Comment by albert-92** [comment]
Just tested it on Windows 10 and Ubuntu 16.04 both with node 6.3.1.

Same problem on both systems:

`MongoDB connection error: MongoError: failed to connect to server [localhost:27017] on first connect
[nodemon] app crashed - waiting for file changes before starting...`

**Comment by woyehenni** [comment]
Mac have the same problem.

**Comment by woyehenni** [comment]
worked by modify gulpfile.babel.js.
// nodemon(`-w ${serverPath} --debug=5858 --debug-brk ${serverPath}`)
nodemon(`-w ${serverPath} --debug=5858 ${serverPath}`)
 but i don't know why this happened.

**Comment by Awk34** [comment]
If you're using a newer version of Node (>= 6.3.0 I think), you can just use the `--inpsect` command: https://nodejs.org/api/debugger.html#debugger_v8_inspector_integration_for_node_js

**Comment by guipsamora** [comment]
I've been constantly running to this issue and I have to restart node using "rs" at least 3 to 4 times, which is making my development pace very slow. 

@woyehenni I've tried what you did but it hasn't worked for me, any other tip or tweaks to make this work on mac? 

Thanks on any input/suggestion

![screen shot 2017-02-12 at 9 53 50 pm](https://cloud.githubusercontent.com/assets/3289179/22867363/d006f83e-f16d-11e6-80af-dc182dc65c9e.png)
